packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/AuthCognito.kt

```diff
@@ -134,13 +138,22 @@ public class AuthCognito : FlutterPlugin, ActivityAware, MethodCallHandler, Plug
   override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
 
     if(call.method == "addPlugin"){
+      if (pluginAdded) {
+        return handleRestartException(
+                HotReloadExceptionCode.PLUGIN_ADDED,
+                result,
+                ExceptionMessages.hotRestartPluginExceptionMessage.format( "Auth" ),
+                ExceptionMessages.hotRestartRecoverySuggestion
+        )
+      }
```

Comment:
I'd like to see if it's possible to avoid expanding our code surface and use the state already maintained by the native libraries.

| Dart | Native | Condition | Action |
| --- | --- | --- | --- |
| No state | No state | Fresh app start | Usual dart validations (plugins before the only one configure) |
| State | State | Hot reload | Have customers use Amplify.isConfigured and skip addPlugins/configure |
| No state | State | Hot restart | Have the native throw their usual exception (Plugin already added, already configured etc) -> In Dart detect this and swallow this exception (only when dart didn't have any state but native did) |
| State | No state | Not possible | Nothing to do |